projects/plugins/jetpack/extensions/blocks/cookie-consent/cookie-consent.php

```diff
@@ -80,3 +82,14 @@ function load_assets( $attr, $content ) {
 	);
 }
 
+/**
+ * Since the cookie consent is part of the cached response HTML, it can still render even when the cookie is set (when it shouldn't).
+ * Because, by default, the cache doesn't vary around the cookie's value.
+ * This let's the cache know that the content has changed to return fresh content.
+ */
+function notify_batcache_that_content_changed() {
+	if ( function_exists( 'vary_cache_on_function' ) ) {
+		// Cast the cookie down to a boolean, to avoid arbitrary code execution.
+		vary_cache_on_function( 'return isset( $_COOKIE[ "' . COOKIE_NAME . '" ] );' );
```

Comment:
For reference, because I didn't know about it:
https://github.com/Automattic/batcache/blob/d4f617b335e9772a61b6d03ad3498b55c8137592/advanced-cache.php#L29